crates/astria-bridge-withdrawer/src/bridge_withdrawer/startup.rs

```diff
@@ -0,0 +1,564 @@
+    let res = tryhard::retry_fn(|| client.tx(tx_hash, false))
+        .with_config(make_cometbft_retry_config(
+            state.clone(),
+            "attempt to get transaction from CometBFT; retrying after backoff".to_string(),
+        ))
+        .await
+        .wrap_err("failed to get transaction from Sequencer after a lot of attempts");
+
+    state.set_sequencer_connected(res.is_ok());
+
+    res
+}
+
+#[instrument(skip_all)]
+async fn get_sequencer_chain_id(
+    client: sequencer_client::HttpClient,
+    state: Arc<State>,
+) -> eyre::Result<tendermint::chain::Id> {
+    let genesis: tendermint::Genesis = tryhard::retry_fn(|| client.genesis())
+        .with_config(make_cometbft_retry_config(
+            state.clone(),
+            "attempt to get genesis from CometBFT; retrying after backoff".to_string(),
+        ))
+        .await
+        .wrap_err("failed to get genesis info from Sequencer after a lot of attempts")?;
+
+    state.set_sequencer_connected(true);
+
+    Ok(genesis.chain_id)
+}
+
+#[instrument(skip_all)]
+async fn get_allowed_fee_asset_ids(
+    client: sequencer_client::HttpClient,
+    state: Arc<State>,
+) -> eyre::Result<AllowedFeeAssetsResponse> {
+    let res = tryhard::retry_fn(|| client.get_allowed_fee_assets())
+        .with_config(make_sequencer_retry_config(
+            state.clone(),
+            "attempt to get allowed fee assets from Sequencer; retrying after backoff".to_string(),
+        ))
+        .await
+        .wrap_err("failed to get allowed fee asset ids from Sequencer after a lot of attempts");
+
+    state.set_sequencer_connected(res.is_ok());
+
+    res
+}
+
+#[instrument(skip_all)]
+async fn get_latest_balance(
+    client: sequencer_client::HttpClient,
+    state: Arc<State>,
+    address: Address,
+) -> eyre::Result<BalanceResponse> {
+    let res = tryhard::retry_fn(|| {
+        // only retry on tendermint_rpc errors, not deserialization or native conversion
+        let client = client.clone();
+        async move {
+            match client.get_latest_balance(address).await {
+                Ok(res) => Ok(Ok(res)),
+                Err(err) => {
+                    if let Some(tendermint_err) = err.as_tendermint_rpc() {
+                        Err(tendermint_err.inner().clone())
+                    } else {
+                        Ok(Err(err))
+                    }
+                }
+            }
+        }
+    })
+    .with_config(make_cometbft_retry_config(
+        state.clone(),
+        "attempt to get latest balance from CometBFT; retrying after backoff".to_string(),
+    ))
+    .await
+    .wrap_err("failed to get latest balance from Sequencer after a lot of attempts");
+
+    let res = res?.wrap_err("failed to deserialize the latest balance response from CometBFT");
+
+    // set cometbft as connected if received a response
+    state.set_sequencer_connected(res.is_ok());
+
+    res
+}
+
+fn make_cometbft_retry_config(
+    state: Arc<State>,
```

Comment:
As for the other one - this is essentially a oneshot method at startup - if it returns you know that you are "connected". Make this simpler be removing it.